src/Protocol/ForgeHandshake.cpp

```diff
@@ -0,0 +1,365 @@
+
+// ForgeHandshake.cpp
+
+// Implements Forge protocol handshaking
+
+#include "Globals.h"
+#include "ForgeHandshake.h"
+#include "json/json.h"
+#include "../Server.h"
+#include "../Bindings/PluginManager.h"
+#include "../ClientHandle.h"
+#include "../Root.h"
+
+
+/** Discriminator byte values prefixing the FML|HS packets to determine their type. */
+enum
+{
+	Discriminator_ServerHello = 0,
+	Discriminator_ClientHello = 1,
+	Discriminator_ModList = 2,
+	Discriminator_RegistryData = 3,
+	Discriminator_HandshakeReset = -2,
+	Discriminator_HandshakeAck = -1,
+};
```

Comment:
Hmm, changing Discriminator to a named enum works in clang but fails with gcc https://travis-ci.org/cuberite/cuberite/jobs/258806610:

/home/travis/build/cuberite/cuberite/src/Protocol/ForgeHandshake.cpp:343:8: error: ‘Discriminator’ is not a class, namespace, or enumeration
   case Discriminator::ClientHello: